diesel_codegen_shared/src/schema_inference/pg.rs

```diff
@@ -2,6 +2,98 @@ use diesel::*;
 use diesel::pg::PgConnection;
 use std::error::Error;
 
+use super::data_structures::*;
+
+table! {
+    pg_attribute (attrelid) {
+        attrelid -> Oid,
+        attname -> VarChar,
+        atttypid -> Oid,
+        attnotnull -> Bool,
+        attnum -> SmallInt,
+        attisdropped -> Bool,
+    }
+}
+
+table! {
+    pg_type (oid) {
+        oid -> Oid,
+        typname -> VarChar,
+    }
+}
+
+joinable!(pg_attribute -> pg_type (atttypid));
+select_column_workaround!(pg_attribute -> pg_type (attrelid, attname, atttypid, attnotnull, attnum, attisdropped));
+select_column_workaround!(pg_type -> pg_attribute (oid, typname));
+
+table! {
+    pg_index (indrelid) {
+        indrelid -> Oid,
+        indexrelid -> Oid,
+        indkey -> Array<SmallInt>,
+        indisprimary -> Bool,
+    }
+}
+
+table! {
+    pg_class (oid) {
+        oid -> Oid,
+        relname -> VarChar,
+    }
+}
+
+pub fn determine_column_type(attr: &ColumnInformation) -> Result<ColumnType, Box<Error>> {
+    let is_array = attr.type_name.starts_with("_");
```

Comment:
Why do array types start with an underscore? This ist just the values of `pg_types.typname`, right? (The [docs](https://www.postgresql.org/docs/9.5/static/catalog-pg-type.html) don't say much about the values in there)